lib/new-settings-loader.js

```diff
@@ -1,71 +1,111 @@
-
-// TODO new-admin save to actual db, like the old settings
-low(adapter).then(it => {
-  db = it
-})
+const NEW_SETTINGS_LOADER_SCHEMA_VERSION = 2
 
 function saveAccounts (accountsToSave) {
-  const currentState = db.getState() || {}
-  const accounts = currentState.accounts || {}
-
-  const newAccounts = _.assign(accounts)(accountsToSave)
+  const sql = `update user_config set data = $2, valid = $3, schema_version = $4 where type = $1;
+  insert into user_config (type, data, valid, schema_version)
+  select $1, $2, $3, $4 where $1 not in (select type from user_config)`
 
-  const newState = _.set('accounts', newAccounts, currentState)
-  db.setState(newState)
-  return db.write()
-    .then(() => newState.accounts)
+  return loadAccounts()
+    .then(currentAccounts => {
+      const newAccounts = _.assign(currentAccounts, accountsToSave)
+      return db.none(sql, ['accounts', { accounts: newAccounts }, true, NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    })
 }
 
-function getAccounts () {
-  const state = db.getState()
-  return state ? state.accounts : null
+function loadAccounts () {
+  const toFields = fieldArr => _.fromPairs(_.map(r => [r.code, r.value], fieldArr))
+  const toPairs = r => [r.code, toFields(r.fields)]
+
+  const sql = `select data
+  from user_config
+  where type=$1
+  and schema_version=$2
+  and valid
+  order by id desc
+  limit 1`
+
+  return db.oneOrNone(sql, ['accounts', NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    .then(row => {
+      if (!row) return {}
+      return _.fromPairs(_.map(toPairs, row.data.accounts))
+    })
 }
 
 function saveConfig (config) {
-  const currentState = db.getState() || {}
-  const currentConfig = currentState.config || {}
-  const newConfig = _.assign(currentConfig, config)
-
-  const newState = _.set('config', newConfig, currentState)
-  db.setState(newState)
-  return db.write()
-    .then(() => newState.config)
-}
+  const sql = 'insert into user_config (type, data, valid, schema_version) values ($1, $2, $3, $4)'
 
-function getConfig () {
-  const state = db.getState()
-  return (state && state.config) || {}
+  return loadLatestConfig()
+    .then(currentConfig => {
+      const newConfig = _.assign(currentConfig, config)
+      return db.none(sql, ['config', { config: newConfig }, true, NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    })
 }
 
 function loadLatest () {
-  return new Promise((resolve) => {
-    if (!db) {
-      setTimeout(() => {
-        return resolve(db.getState())
-      }, 1000)
-    } else {
-      return resolve(db.getState())
-    }
-  })
+  return Promise.all([loadLatestConfig(), loadAccounts()])
+    .then(([config, accounts]) => ({
+      config,
+      accounts
+    }))
+}
+
+function loadLatestConfig () {
+  const sql = `select data
+  from user_config
+  where type=$1
+  and schema_version=$2
+  and valid
+  order by id desc
+  limit 1`
+
+  return db.oneOrNone(sql, ['config', NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    .then(row => !row ? {} : row.data.config)
```

Comment:
Is it possible for the query to return a null response? Isn't this QueryResultError already?

Also we don't want to return a empty object for config, just throw the error if we dont find a config.